core-meta.html

```diff
@@ -92,24 +134,39 @@
         return metaData[t];
       },
 
-      register: function(id, old) {
-        if (id && id !== SKIP_ID) {
-          this.unregister(this, old);
-          this.metaData[id] = this;
-          this.metaArray.push(this);
+      resetRegistration: function (oldKey, oldValue) {
+        this.unregister(oldKey, oldValue);
+        this.registerSelf();
+      },
+
+      unregisterSelf: function () {
+        this.unregister(this.key, this.value);
+      },
+
+      registerSelf: function() {
+        var key = this.key;
+        if (key !== '' && key !== SKIP_ID) {
+          this.metaData[key] = this.value;
+          this.metaArray.push(this.value);
         }
       },
 
-      unregister: function(meta, id) {
-        delete this.metaData[id || meta.id];
-        var i = this.metaArray.indexOf(meta);
-        if (i >= 0) {
-          this.metaArray.splice(i, 1);
+      unregister: function(key, value) {
```

Comment:
I considered that. It has been treated as de-facto internal implementation prior to my change, so I wasn't sure if it needed default behavior. I can go either way.